cpp/arcticdb/version/version_core.cpp

```diff
@@ -1426,16 +1496,21 @@ VersionedItem sort_merge_impl(
                 },
                 DynamicSchema{pipeline_context->descriptor(), index},
                 [pipeline_context=pipeline_context, &fut_vec, &store](SegmentInMemory &&segment) {
-                    auto local_index_start = TimeseriesIndex::start_value_for_segment(segment);
-                    auto local_index_end = TimeseriesIndex::end_value_for_segment(segment);
+                    const auto local_index_start = TimeseriesIndex::start_value_for_segment(segment);
+                    const auto local_index_end = TimeseriesIndex::end_value_for_segment(segment);
+                    for (const std::shared_ptr<Column>& col : segment.columns()) {
+                        if (col->row_count() == 0) {
+                            col->default_initialize_rows(0, segment.row_count(), true);
```

Comment:
I think think that this should be handled by a sparse_map but I could find how to make it work. I tried calling set_sparse_map and append_sparse_map but it didn't work.

Reply:
Surely a staged segment with 0 rows always be compacted together with other incomplete segments? We're never going to write a data key with zero rows here, that would be pointless

Reply:
Yeah, but imagine the situation where you have a dynamic schema.
Staged segment 1 has columns [a, b] and 0 rows
Staged segment 2 has columns[a] and 10 rows
It would column b into the descriptor and column b must have 10 rows. Currently they are default initialized but it would be cool if it could have sparse map (or empty type at some point)